core/src/main/java/org/frankframework/pipes/RegExPipe.java

```diff
@@ -0,0 +1,202 @@
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package org.frankframework.pipes;
+
+import lombok.Getter;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.PipeForward;
+import org.frankframework.core.PipeLineSession;
+import org.frankframework.core.PipeRunException;
+import org.frankframework.core.PipeRunResult;
+import org.frankframework.doc.Forward;
+import org.frankframework.doc.Mandatory;
+import org.frankframework.stream.Message;
+import org.frankframework.util.XmlBuilder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+/**
+ * Tries to match the input against the provided regex.
+ *
+ * <pre>{@code
+ * 	<RegExPipe name="regExPipe" regex="^(.*?)(name!)$" flags="CASE_INSENSITIVE">
+ * 		<Forward name="then" path="EXIT" />
+ * 		<Forward name="else" path="EXIT" />
+ * 	</RegExPipe>
+ * }</pre>
+ *
+ * <p>Input:</p>
+ * <pre>{@code
+ * 	Hello name!
+ * }</pre>
+ *
+ * <p>Output:</p>
+ * <pre>{@code
+ *  <matches>
+ * 		<match index="1" value="Hello name!">
+ * 			<group index="1">Hello </group>
+ * 			<group index="2">name!</group>
+ * 		</match>
+ *  </matches>
+ * }</pre>
+ *
+ * @ff.tip https://regex101.com can be used to quickly create and debug regex expressions.
+ *
+ */
+@Forward(name = RegExPipe.THEN_FORWARD, description = "When a match is found.")
+@Forward(name = RegExPipe.ELSE_FORWARD, description = "When no match is found.")
+public class RegExPipe extends AbstractPipe {
+
+	protected final static String THEN_FORWARD = "then";
+	protected final static String ELSE_FORWARD = "else";
+
+	public enum RegExFlag {
+		CANON_EQ(Pattern.CANON_EQ),
+		CASE_INSENSITIVE(Pattern.CASE_INSENSITIVE),
+		COMMENTS(Pattern.COMMENTS),
+		DOT_ALL(Pattern.DOTALL),
+		LITERAL(Pattern.LITERAL),
+		MULTILINE(Pattern.MULTILINE),
+		UNICODE_CASE(Pattern.UNICODE_CASE),
+		UNICODE_CHARACTER_CLASS(Pattern.UNICODE_CHARACTER_CLASS),
+		UNIX_LINES(Pattern.UNIX_LINES);
+
+		private final @Getter int flag;
+
+		RegExFlag(int flag) {
+			this.flag = flag;
+		}
+	}
+
+	private String regex = "";
+	private List<RegExFlag> flags = List.of();
+
+	private Pattern pattern;
+
+	private PipeForward thenForward;
+	private PipeForward elseForward;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+
+		final int flags = this.flags.stream()
+				.map(RegExFlag::getFlag)
+				.reduce(0, (a, b) -> a | b);
+
+		if (this.regex == null || this.regex.isEmpty()) {
+			throw new ConfigurationException("Value for `regex` is mandatory.");
+		}
+
+		try {
+			this.pattern = Pattern.compile(this.regex, flags);
+		} catch (PatternSyntaxException e) {
+			throw new ConfigurationException(e.getMessage());
```

Comment:
Het is denk ik beter om 
```suggestion
			throw new ConfigurationException("duidelijke melding voor de eindgebruiken", excpetie voor de log file);
```
te gebruiken. Dat helpt on den duur ons alsook de gebruiker die meer heeft aan 'de regex klopt niet' dan wat hier uit mag komen.